.github/workflows/ci.yml

```diff
@@ -0,0 +1,13 @@
+name: "CI"
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```

Comment:
A deprecation warning is thrown after the pre-commit job has run:
https://github.com/wikimedia/liftwing-python/actions/runs/9481533171

I looked at the affected actions and they have new versions we could use:
1. checkout: https://github.com/actions/checkout?tab=readme-ov-file#usage
2. setup-python: https://github.com/actions/setup-python?tab=readme-ov-file#basic-usage

Reply:
Nice catch! I have updated the versions and also added python version >= 3.9 as required.